src/backend/joanie/core/models/products.py

```diff
@@ -957,17 +958,39 @@ def submit_for_signature(self, user: User):
             name=contract_definition.name, context=context
         )
 
-        was_already_submitted = (
+        was_already_submitted = bool(
             contract.submitted_for_signature_on and contract.signature_backend_reference
         )
-        should_be_resubmitted = was_already_submitted and (
+
+        should_be_resubmitted = was_already_submitted and bool(
             not contract.is_eligible_for_signing() or contract.context != context
         )
 
         if should_be_resubmitted:
-            backend_signature.delete_signing_procedure(
-                contract.signature_backend_reference
-            )
+            # The signature provider may take some time to respond with the confirmation of the
+            # deletion. If we reach the timeout limit, we should reset the contract submission
+            # values because the signature provider will delete them.
+            # We won't be able to use the `contract.signature_backend_reference` again.
+            try:
+                backend_signature.delete_signing_procedure(
+                    contract.signature_backend_reference
+                )
+            except ReadTimeout as exception:  # pylint: disable=unused-variable
+                message = (
+                    "Signature Provider is taking a while on deletion "
+                    f"of reference {contract.signature_backend_reference}."
+                )
+                logger.error(
+                    message,
+                    extra={
+                        "context": {
+                            "order": self.to_dict(),
+                            "product": self.product.to_dict(),
+                        }
+                    },
+                )
```

Comment:
I think we could simplify this:
```suggestion
                logger.error(
                    "Timeout on signature reference deletion",
                    extra={
                        "context": {
                            "order": self.to_dict(),
                            "product": self.product.to_dict(),
                            "signature_backend_reference": contract.signature_backend_reference,
                        }
                    },
                )
```
If I'm not mistaking, using a common message would allow sentry to group all those events together.

Reply:
thank you :pray: